### PR TITLE
Add open graph meta tags

### PR DESCRIPTION
**Pull request by attfarhan**
Adds open graph meta tags to prettify sharing on social media.  Produces cards with logo and generic description and title for all links.  
Facebook: http://cl.ly/3j0o422M293O
Twitter: http://cl.ly/1S321R2j2D3i
Slack: http://cl.ly/030x100T2e3q

##### Reviewer tasks

- [x] HACKS: reviewer approves hacks introduced by this change
- [x] DOCS: reviewer approves the docstrings (or the justified omission thereof)
- [x] UNIT-TEST: reviewer approves the unit tests (or the justified omission thereof)

### CODE REVIEW

File: app/templates/layout.html [hunk 1]
```diff
@@ -11,6 +11,18 @@
...
+		<link rel="image_src" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
+		<meta property="og:title" content="Sourcegraph">
+		<meta property="og:image" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
+		<meta property="og:type" content="website" />
+		<meta property="og:description" content="Search for a function, type, or package, and see how other developers use it, globally." />
+		<meta name="twitter:card" content="summary" />
+		<meta name="twitter:site" content="Sourcegraph" />
+		<meta name="twitter:title" content="Sourcegraph" />
+		<meta name="twitter:image" content="https://pbs.twimg.com/profile_images/698927855237922817/5qzuSmD1.png" />
+		<meta name="twitter:description" content="Search for a function, type, or package, and see how other developers use it, globally." />
```
Comment (sqs) [inline app/templates/layout.html]: The 2 descriptions should be:

> How developers discover and understand code. Sourcegraph is a fast, global, semantic code search and cross-reference engine.

File: app/templates/layout.html [hunk 2]
```diff
@@ -11,6 +11,18 @@
 		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
 		<meta name="theme-color" content="#0fb6f2">
 
+		{{/* Social */}}
+		<link rel="image_src" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
+		<meta property="og:title" content="Sourcegraph">
+		<meta property="og:image" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
```
Comment (sqs) [inline app/templates/layout.html]: Do not use content hosted on GitHub. That lets them track all of our traffic. Instead use the .png files hosted on sourcegraph.com, or upload them to an S3 bucket.

File: app/templates/layout.html [hunk 3]
```diff
@@ -11,6 +11,18 @@
...
+		<meta property="og:type" content="website" />
+		<meta property="og:description" content="Search for a function, type, or package, and see how other developers use it, globally." />
+		<meta name="twitter:card" content="summary" />
+		<meta name="twitter:site" content="Sourcegraph" />
+		<meta name="twitter:title" content="Sourcegraph" />
+		<meta name="twitter:image" content="https://pbs.twimg.com/profile_images/698927855237922817/5qzuSmD1.png" />
```
Comment (sqs) [inline app/templates/layout.html]: This is just our logo, so use the same one as you'd use for the image_src and og:image ones above.

File: app/templates/layout.html [hunk 4]
```diff
@@ -11,6 +11,18 @@
 		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
 		<meta name="theme-color" content="#0fb6f2">
 
+		{{/* Social */}}
+		<link rel="image_src" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
+		<meta property="og:title" content="Sourcegraph">
```
Comment (sqs) [inline app/templates/layout.html]: Be consistent about using or not using self-closing tags (the other meta tags have `/>`). For meta tags, I would say don't close them.

File: app/templates/layout.html (hunk 2 above)
Comment (sqs) [inline app/templates/layout.html]: Escape ampersand as `&amp;`

File: app/templates/layout.html [hunk 5]
```diff
@@ -11,6 +11,18 @@
 		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
 		<meta name="theme-color" content="#0fb6f2">
 
+		{{/* Social */}}
+		<link rel="image_src" content="https://avatars2.githubusercontent.com/u/3979584?v=3&s=200" />
```
Comment (sqs) [inline app/templates/layout.html]: Escape ampersand as `&amp;`

**Comment by attfarhan** [comment]
This has been updated.

File: app/misc_handlers.go [hunk 6]
```diff
@@ -23,15 +23,15 @@ func robotsTxt(w http.ResponseWriter, r *http.Request) {
 func robotsTxtHelper(w io.Writer, allowRobots bool, sitemapUrl string) {
 	var buf bytes.Buffer
 	fmt.Fprintln(&buf, "User-agent: *")
-	if allowRobots {
-		fmt.Fprintln(&buf, "Allow: /")
-		// Disallow URLs with revisions
-		fmt.Fprintln(&buf, "Disallow: /*/*/*@*/")
-		fmt.Fprintln(&buf, "Disallow: /*/*@*/")
+	// if allowRobots {
```
Comment (sqs) [inline app/misc_handlers.go]: Is this intentional?

File: app/templates/layout.html [hunk 7]
```diff
@@ -11,6 +11,20 @@
 		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
 		<meta name="theme-color" content="#0fb6f2">
 
+		{{/* Social */}}
```
Comment (sqs) [inline app/templates/layout.html]: Unnecessary comment

File: app/templates/layout.html [hunk 8]
```diff
@@ -11,6 +11,20 @@
...
+		<meta property="og:title" content="Sourcegraph">
+		<meta property="og:image" content="https://sourcegraph.com/.assets/v/16483_2016-06-03_27d4bc2/img/sourcegraph-mark.png">
+		<meta property="og:image:height" content="90">
+		<meta property="og:image:width" content="90">
+		<meta property="og:type" content="website">
+		<meta property="og:description" content="How developers discover and understand code. Sourcegraph is a fast, global, semantic code search and cross-reference engine.">
+		<meta name="twitter:card" content="summary">
+		<meta name="twitter:site" content="Sourcegraph">
+		<meta name="twitter:title" content="Sourcegraph">
+		<meta name="twitter:image" content="https://sourcegraph.com/.assets/v/16483_2016-06-03_27d4bc2/img/sourcegraph-mark.png">
```
Comment (sqs) [inline app/templates/layout.html]: Use `{{assetURL "/img/sourcegraph-mark.png"}}` to produce stable URLs. These hardcoded URLs have the commit ID in them, so I don't think they'll still work after a subsequent deploy of a new commit ID.

**Comment by rothfels** [comment]
LGTM

**Pull request closed without merging**